pretyping/pretyping.ml

```diff
@@ -999,6 +999,24 @@ struct
           in
           sigma, body, na, c1, Esubst.subs_id 0, c2, trace
         in
+        let sigma, template, c1 = match template with
+          | None | Some ([],_) -> sigma, template, c1
+          | Some (None :: t, bound) -> sigma, Some (t,bound), c1
+          | Some (Some u0 :: t, bound) ->
+            (* template arg: type guaranteed to be syntactically an arity
+               we replace the output universe with a fresh one *)
+            let sigma, u = Evd.new_univ_level_variable UState.univ_flexible_alg sigma in
```

Comment:
>    Small example reduced from fcl:

We could also try 
```suggestion
            let sigma, u = Evd.new_univ_level_variable UState.univ_rigid sigma in
```
here but I can't really predict the consequences

Reply:
This looks wrong, why would this be rigid when the user has no control over which type is generated here?

Reply:
It's rather adhoc but maybe that's normal when trying to get minimization heuritics to be backwards compatible.

Reply:
Let's try it on CI

Reply:
(test suite failures are not concerning)